### DISCUSSION TITLE
Default values not respected in sqlalchemy

### DISCUSSION THREAD

**Discussion by hasanparasteh** [sqlalchemy/sqlalchemy#10846] [Usage Questions] [answered]
### Describe the bug

I setup my environment to use SQLAlchemy for my next project but then something wrong happens. i use `fastapi` example to get started and then tried to add some default values to Column declaration. 

here is my table example:
```py
marketers = sqlalchemy.Table(
    'Users',
    metadata,

    Column('id', Integer, primary_key=True, index=True),
    Column('username', String(32), index=True, unique=True),
    Column('password', String(255)),

    Column('email', String(255), unique=True, index=True, nullable=True),
    Column('phone', String(11), unique=True, index=True, nullable=True),

    Column('create_time', DOUBLE, default=datetime.datetime.now().timestamp()),
    Column('update_time', DOUBLE, default=datetime.datetime.now().timestamp()),

    Column('is_active', Boolean, default=True),
)
```

The problem is that when i call this method only username and password will be inserted and `create_time` & `update_time` and is_active will be inserted as `null`.

fastapi example: [refrence](https://fastapi.tiangolo.com/how-to/async-sql-encode-databases/#import-and-set-up-sqlalchemy)

### SQLAlchemy Version in Use

1.4.42

### DBAPI (i.e. the database driver)

pymysql

### Database Vendor and Major Version

mysql 8

### Python Version

3.10

### Operating system

OSX

### To Reproduce

```python
@router.post("/")
async def create_marketer(marketer: NewMarketer):
    hashed_password = bcrypt.kdf(
        bytes(marketer.password, 'utf-8'),
        desired_key_bytes=32,
        rounds=100,
        salt=bytes(config.PASSWORD_SALT, 'utf-8')
    )
    query = Marketers.create().values(username=marketer.username, password=hashed_password.hex())
    print(query)
    try:
        last_record_id = await database.execute(query)
        return {'result': True, 'id': last_record_id}
    except pymysql.IntegrityError:
        raise HTTPException(status_code=422, detail='username is already full')
```

### Error

```
Default values wont be inserted.
``` 

### Additional context

Maybe I'm misusing the API. but can you guys detect how this happens? and how can i fix it?

**Comment by zzzeek** [comment, marked as answer]
your code sample shows a `Table` construct but then later you show some entirely different API that is specific to your application and is not shown ("`Marketers.create().values(...)`") so we would assume that however this Table is mapped to that `create().values()` set of functions is incorrectly overwriting these attributes.

Also there's a critical mistake in your "updated_at" and "created_at" defaults which is that you've hardcoded a single time value in your table metadata, you want those to be callables:

```py
marketers = sqlalchemy.Table(
    'Users',
    # ...

    Column('create_time', DOUBLE, default=lambda: datetime.datetime.now().timestamp()),
    Column('update_time', DOUBLE, default=lambda: datetime.datetime.now().timestamp()),

)

```
